### DISCUSSION TITLE
Is there any client for Angular 12 app?

### DISCUSSION THREAD

**Discussion by ASolod82** [open-policy-agent/community#71] [Community]
Hi Guys,

First of all frankly speaking, I tried to google something to query OPA from my angular application. There is some crumbs of information on youtube from Chef, an express framework example on github in zero-star repo, and that's it :( 

Can anyone advise some npm library, some best practice example for angular app? 

Thanks in Advance
Artöm

**Comment by srenatus** [comment]
> There is some crumbs of information on youtube from Chef

The code that was talked about there is open source. However, it's a pretty special mix, including an ["authz-service"](https://github.com/chef/automate/tree/main/components/authz-service) that's wrapping OPA as a Go library, and a small [API gateway](https://github.com/chef/automate/tree/main/components/automate-gateway); also the [UI code](https://github.com/chef/automate/tree/main/components/automate-ui). The `<authorized>` component could be a starting point for digging into the UI side of things, [see here](https://github.com/chef/automate/blob/main/components/automate-ui/src/app/components/authorized/authorized.component.ts).

But yeah, it's not communicating with the OPA API, since that isn't exposed in that system.
